### DISCUSSION TITLE
Disable Warning screen printouts for setup and compare_models?

### DISCUSSION THREAD

**Discussion by wodecki** [pycaret/pycaret#1951] [Q&A]
Hi, 

I want to disable warning printouts displayed on-screen during a python script (not Jupyter Notebook) execution of a classification setup and compare_models.

I tried many different techniques, including warnings simplefilter, filterwarnings, custom redirection with contextmanager, etc. but those printouts always appear on my screen. 

Any idea how to suppress them?

Many thanks in advance :)

Andy

**Comment by Ruibin-Liu** [comment]
[This one](https://stackoverflow.com/a/55595680) works for me. Be careful though because it suppresses all warnings. Maybe only uncomment these lines when you know what's really going on.
```
if not sys.warnoptions:
    warnings.simplefilter("ignore")
    os.environ["PYTHONWARNINGS"] = "ignore" # Also affect subprocesses
```
